### DISCUSSION TITLE
Can't get results of inner join

### DISCUSSION THREAD

**Discussion by Lana-L** [prisma/prisma#17183] [Q&A] [answered]
Hi
I am very new to all this and I thought I followed the instructions in the doc, but what I'm getting is not what I expect. 
I am expecting this behaviour (I tested it in my db and it works) - `select * from meal inner join recipes on meal.meal_id = recipes.meal_id`

this is the model I have
```
model meal {
  meal_id   Int       @id @default(autoincrement())
  name      String?
  cook_time String?
  recipes   recipes[]
}

model recipes {
  recipe_id    Int     @id @default(autoincrement())
  servings     Int?
  instructions String?
  meal_id      Int
  meal         meal   @relation(fields: [meal_id], references: [meal_id])
}
```

this is the query I have
```ts
 const meals = await prisma.meal.findMany({
        include: {
            recipes:  true
                }
    })
```
I get this as an example when I console.log(meals)

```ts
{
    meal_id: 74,
    name: 'Cheesy Chicken Noodle Casserole',
    cook_time: '35M',
    recipes: [ [Object] ]
  }
```
ie, I cannot access anything in the recipes table. 

when I enable logging on prisma, i see this, which is very confusing.
```
prisma:query SELECT `main`.`meal`.`meal_id`, `main`.`meal`.`name`, `main`.`meal`.`cook_time` FROM `main`.`meal` WHERE 1=1 LIMIT ? OFFSET ?
prisma:query SELECT `main`.`recipes`.`recipe_id`, `main`.`recipes`.`servings`, `main`.`recipes`.`instructions`, `main`.`recipes`.`meal_id` FROM `main`.`recipes` WHERE `main`.`recipes`.`meal_id` IN (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?) LIMIT ? OFFSET ?
prisma:query SELECT `main`.`cuisines`.`cuisine_id`, `main`.`cuisines`.`name` FROM `main`.`cuisines` WHERE 1=1 LIMIT ? OFFSET ?
```
any suggestions of what I'm doing incorrectly?

**Comment by janpio** [comment, marked as answer]
`console.log` by default only logs 1 level of the returned data - which is why is says `[Object]` instead of showing the list of recipes. When you do `console.log(meals.recipes)` it will show all the data you requested.

**Reply by Lana-L** [reply]
Hi
if I console.log(meals.recipes), I get "undefined" (which is what I was getting in my app and tried to console.log to troubleshoot; I then also found the prisma logging and found the odd expression above).

I even console.log(1) to make sure I know where it appears:
```ts
console.log(1);
    console.log(meals.recipes);
```
```
1
undefined
```

**Reply by janpio** [reply]
Argh, I messed up. `meals` is a list of _all_ meals of course - sorry. You need to loop through them and log each individual `meal.recipes`. Or if you are confident the first returned meal has recipes, you can also `console.log(meals[0].recipes)` to log the recipes of the first entry in the result.

**Reply by Lana-L** [reply]
thank you so much, it's now working :)

i looped through it in the app like this and it's working as intended

```html
<tbody>
      {#each data.meals as meal}
           <tr>
              <td>{meal.name}</td>
                   <td>{meal.cook_time}</td>
                    {#each meal.recipes as recipe}
                        <td>{recipe.instructions}</td>
                    {/each}
            </tr>
       {/each}
</tbody>
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
